airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/MergeUnions.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+class MergeUnions : AirbyteSchemaIdentityMapper {
+    override fun mapUnion(schema: UnionType): AirbyteType {
+        // Map the options first so they're in their final form
+        val mappedOptions = schema.options.map { map(it) }
```

Comment:
This would only recurse infinitely if an option contained a reference to the top level schema. It is technically possible to construct this in code, but it is not possible to represent it in json.